app/routes/github.js

```diff
@@ -28,30 +28,30 @@ export default Route.extend(AuthenticatedRouteMixin, {
     let stateValidator = get(this, 'githubState');
 
     if (stateValidator.validate(state)) {
-      return this._sendRequest(code);
+      return this._sendRequest(code, state);
```

Comment:
The `github-state` service perform the validation locally, using the session data facilities provided by ember-simple-auth, so at the moment, there is no need to send the state to the API.

We discussed that as an option, but I don't think it's MVP and plenty of other libraries do such validation locally. I don't think it falls into the "should our client talk to github directly or through our API" story.